dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/DataRequestManager.kt

```diff
@@ -117,21 +137,147 @@ class DataRequestManager(
         return aggregatedDataRequests
     }
 
+    /**
+     * Method to retrieve a data request by its ID
+     * @param dataRequestId the ID of the data request to retrieve
+     * @return the data request corresponding to the provided ID
+     */
+    @Transactional
+    fun getDataRequestById(dataRequestId: String): StoredDataRequest {
+        throwResourceNotFoundExceptionIfDataRequestIdUnknown(dataRequestId)
+        val dataRequestEntity = dataRequestRepository.findById(dataRequestId).get()
+        return buildStoredDataRequestFromDataRequestEntity(dataRequestEntity)
+    }
+
+    /**
+     * Method to patch the status of a data request.
+     * @param dataRequestId the id of the data request to patch
+     * @param requestStatus the status to apply to the data request
+     * @return the updated data request object
+     */
+    @Transactional
+    fun patchDataRequest(dataRequestId: String, requestStatus: RequestStatus): StoredDataRequest {
+        throwResourceNotFoundExceptionIfDataRequestIdUnknown(dataRequestId)
+        var dataRequestEntity = dataRequestRepository.findById(dataRequestId).get()
+        logger.info("Patching Company ${dataRequestEntity.dataRequestId} with status $requestStatus")
+        dataRequestEntity.requestStatus = requestStatus
+        dataRequestRepository.save(dataRequestEntity)
+        dataRequestEntity = dataRequestRepository.findById(dataRequestId).get()
```

Comment:
On it

Reply:
In my view, it would be better to leave the current code with this 2-line code duplication as is, since this is the only occurrence and outsourcing these two lines would complicate the reading of the function `patchDataRequest` where lines 161 & 165 are identical. However, sonar forced me to refactor the DataRequestManager: now there are two files, a BulkDataRequestManager and a SingleDataRequestManager together with a utils file that contains functions used in both managers

Reply:
ok